### PR TITLE
Video listing and single video page

### PR DESCRIPTION
**Pull request by anuj-kumary**
This PR includes:-

   - Single Video Page
   - Video data updated with required properties
   - Video Listing Page
   - Navbar Componetns
   - Sidebar Componetns

Preview:- https://deploy-preview-2--techhub-react.netlify.app/

### CODE REVIEW

**Review by mohammadkhizerkhan — APPROVED** [review]
-good folder structure
-clean code, Easily understandable

**Review by vinc3nati — APPROVED** [review]
UI also looks great
PR is ready to merge.

File: src/pages/SingleVideo/SingleVideo.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { useParams } from 'react-router-dom';
+import { Sidebar } from '../../components';
+import { useData } from '../../contexts/data-context';
+import './SingleVideo.css';
+
+export const SingleVideo = () => {
+  const { videoID } = useParams();
+  const { state } = useData();
+  const findVideo = state.videos?.find((element) => element._id === videoID);
+  console.log(findVideo);
```
Comment (znjs) [inline src/pages/SingleVideo/SingleVideo.jsx]: can remove the `console.log`

File: src/App.js [hunk 2]
```diff
@@ -1,10 +1,21 @@
+import { Route, Routes } from 'react-router-dom';
 import './App.css';
-import { LandingPage } from './pages/Landing-Page/LandingPage';
+import { Navbar } from './components';
+import Mockman from 'mockman-js';
+import { LandingPage } from './pages';
+import { VideoListing } from './pages';
+import { SingleVideo } from './pages';
```
Comment (znjs) [inline src/App.js]: can import all pages in a single line

File: src/components/Navbar/Navbar.css [hunk 3]
```diff
@@ -0,0 +1,65 @@
+body {
+    width: 100vw;
```
Comment (znjs) [inline src/components/Navbar/Navbar.css:2]: why mentioning specifically its the default behaviour

File: src/components/Sidebar/Sidebar.css [hunk 4]
```diff
@@ -0,0 +1,20 @@
+.sidebar {
+    background-color: var(--bg-color);
+    position: sticky;
+    min-width: 13rem;
+    height: 88vh;
```
Comment (znjs) [inline src/components/Sidebar/Sidebar.css:5]: `position: sticky` won't be effective until a atleast of the top bottom left right is applied

File: src/pages/VideoListing/VideoListing.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Sidebar } from '../../components';
+import { useData } from '../../contexts/data-context';
+import { VideoCard } from './components/VideoCard';
```
Comment (znjs) [inline src/pages/VideoListing/VideoListing.jsx:4]: can `import` directly from components since you defined an `index.js` file

**Review by znjs — APPROVED** [review]
Good folder structure
Good naming conventions
if possible make the suggested changes

**Comment by znjs** [comment]
image
side nav is not sticky

**Comment by znjs** [comment]
image
include `_redirects` file to solved the reload issue
```js
/* /index.html 200
```

File: src/pages/SingleVideo/SingleVideo.jsx (hunk 1 above)
Comment (anuj-kumary) [reply src/pages/SingleVideo/SingleVideo.jsx]: Done

File: src/App.js (hunk 2 above)
Comment (anuj-kumary) [reply src/App.js]: Done

File: src/components/Navbar/Navbar.css (hunk 3 above)
Comment (anuj-kumary) [reply src/components/Navbar/Navbar.css:2]: So this **width: 100vw** for my navbar which is coming from CL I gave width 80vw and here I want 100vw

File: src/components/Sidebar/Sidebar.css (hunk 4 above)
Comment (anuj-kumary) [reply src/components/Sidebar/Sidebar.css:5]: I'll make changes

File: src/pages/VideoListing/VideoListing.jsx (hunk 5 above)
Comment (anuj-kumary) [reply src/pages/VideoListing/VideoListing.jsx:4]: So this VideoCard is inside pages VideoListing components so because of that import like this

**Pull request merged**